plugins/Actions/API.php

```diff
@@ -423,7 +423,14 @@ protected function doFilterPageDatatableSearch($callBackParameters, $table, $sea
             }
 
             // nothing found in all sub tables
-            return new DataTable;
+            $result = new DataTable;
+            $subTables = $table->getDataTables();
+            if (count($subTables) > 0) {
+                // use the first subtable's metadata to ensure basic metadata like `period` is available in response
+                $subTable = reset($subTables);
+                $result->setAllTableMetadata($subTable->getAllTableMetadata());
+            }
+            return $result;
```

Comment:
Will this actually set the correct period though? If we query with a multi period, then this will set it to the first subperiod, so in the last7 example, it would be 7 days ago. Is this what callers of the API would expect? I think maybe the Rss formatter doesn't support multiperiod requests, and some other changes are needed?

Reply:
Actually no. That piece of code is executed for datatable maps inside a map. I'll remove some useless code so it gets more clear.